src/config.rs

```diff
@@ -232,6 +232,10 @@ pub struct CompactorOptions {
     /// An optional tokio runtime handle to use for scheduling compaction work. You can use
     /// this to isolate compactions to a dedicated thread pool.
     pub compaction_runtime: Option<Handle>,
+
+    /// An optional filter that can perform filtering and modification on rows
+    /// during the compaction operation.
+    pub compaction_filter: Arc<Box<dyn CompactionFilter>>,
```

Comment:
A `Vec<>` of this would allow filter chaining, which I think we want (based on your comments and my response [here](https://github.com/slatedb/slatedb/issues/225#issuecomment-2369073728))